File: value/src/main/java/com/bnorm/barkeep/model/ComponentValue.java
// Copyright 2016 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.model;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class ComponentValue implements Component {

  public static Builder builder() {
    return new AutoValue_ComponentValue.Builder();
  }

  @AutoValue.Builder
  abstract static class Builder {

    abstract ComponentValue build();

    abstract Builder setIngredient(Ingredient ingredient);

    abstract Builder setMin(double min);

    abstract Builder setMax(Double max);

    abstract Builder setComponentNum(long componentNum);

    abstract Builder setOrder(long order);
  }
}

File: model-value/src/main/kotlin/com/bnorm/barkeep/model/value/ComponentValue.kt
// Copyright 2017 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.model.value

import com.bnorm.barkeep.model.Component
import com.bnorm.barkeep.model.Ingredient
import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

data class ComponentValue(override val ingredient: Ingredient,
                          override val min: Double,
                          override val max: Double? = null,
                          override val componentNum: Long,
                          override val order: Long) : Component

object ComponentValueAdapter {
  @ToJson fun toJson(component: Component): ComponentValue {
    return ComponentValue(component.ingredient, component.min, component.max, component.componentNum, component.order)
  }

  @FromJson fun fromJson(json: ComponentValue): Component {
    return json
  }
}
